### PR TITLE
Playlists, Livestreams, Podcasts

### PR DESCRIPTION
**Pull request by flipreverse**
This is a first draft. It add support to set m3u playlists as action on cards or directly play them.
At the moment, one is *not* able to edit cards assigned to a m3u playlist. This is due a `None` answer by `get_song_by_url()` in the player.
Why is this function called on behalf of the HTTP request `#/cards/0006807296/edit`?

### CODE REVIEW

**Comment by s-martin** [comment]
Thanks for the pull request 🎉👍

At the moment I'm unsure what the PR is adding, because according to our [feature status page](https://github.com/MiczFlor/RPi-Jukebox-RFID/blob/future3%2Fdevelop/documentation%2Fdevelopers%2Fstatus.md) playlists should work already. But I haven't tried playlists myself.

**Comment by flipreverse** [comment]
> Thanks for the pull request 🎉👍
> 
> At the moment I'm unsure what the PR is adding, because according to our [feature status page](https://github.com/MiczFlor/RPi-Jukebox-RFID/blob/future3%2Fdevelop/documentation%2Fdevelopers%2Fstatus.md) playlists should work already. But I haven't tried playlists myself.

Yes and no. At the moment, jukebox-daemon reads playlists, and presents the content as single items to the user. See `playlistgenerator.py:206`. With my patch, playlists are displayed as m3u files in the selection dialogue. Moreover, when selected by a dialogue or via a card MPD loads the playlist. See line 422 in `src/jukebox/components/playermpd/__init__.py)`.

**Comment by pabera** [comment]
For now, playlists haven't been handled properly. If we want to add this feature, I suggest get them running everywhere, Web App (e.g. displaying the current playlist, aka Queue), Cards, Player etc.

**Comment by pabera** [comment]
@flipreverse 
It took me a bit of time to dive into playlists and livestreams. I remembered that most of it was implemented, but since I don't use it myself and it was not documented, I didn't remember if there were any gaps. 

Good news: Playlists are implemented, even without your change. They also can be assigned to cards. The problem is, how it works has not been described well. 

I will use your PR to make the necessary changes to the docs and revert your change as it actually conflicts with the original implementation. Here is an explanation from the code that already describes how to use playlists.

Long story short. The Phoniebox expands the content of your playlist, so you can actually call individual items from it within the Web App for example. To be able to assign a play list to a card, I suggest the following:

1. Create a new folder.
2. Place your single *.m3u file into this folder
3. Register the folder to your card. 

This will play your playlist. But instead of registering a playlist to a card, you register a folder and Phoniebox will do the rest.

https://github.com/MiczFlor/RPi-Jukebox-RFID/blob/4f015dde6b348dadf0ebf5dae2de89ba9cf18046/src/jukebox/jukebox/playlistgenerator.py#L2-L32

**Comment by s-martin** [comment]
> https://github.com/MiczFlor/RPi-Jukebox-RFID/blob/4f015dde6b348dadf0ebf5dae2de89ba9cf18046/src/jukebox/jukebox/playlistgenerator.py#L2-L32

I guess it could make sense to pull some docs from docstring to the Markdown docs or at least reference it. E.g. like this  https://github.com/MiczFlor/RPi-Jukebox-RFID/blob/6f0976443980d787473085fc9d8feedd659280b4/documentation/developers/README.md?plain=1#L13-L14

Battery Monitor docs are also such a candidate.

File: documentation/builders/webapp/playlists-livestreams-podcasts.md [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+```
+* audiofolders
+  * wake-up-songs
+    playlist.m3u
+  * lullabies-sleep-well
+    playlist.m3u
+```
+
+### Assiging a .m3u playlist to a card
```
Comment (s-martin) [inline documentation/builders/webapp/playlists-livestreams-podcasts.md:52]:
```suggestion
### Assigning a .m3u playlist to a card
```

File: documentation/builders/webapp/playlists-livestreams-podcasts.md [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+#### Example of livestream.txt
+
+```txt
+https://wdr-diemaus-live.icecastssl.wdr.de/wdr/diemaus/live/mp3/128/stream.mp3
+http://channels.webradio.antenne.de/hits-fuer-kids
+```
+
+## Podcasts
+
+Just like you add livestreams to the Phoniebox, you can also add indivdual Podcasts or entire Podcast feeds to the Phoniebox.
```
Comment (s-martin) [inline documentation/builders/webapp/playlists-livestreams-podcasts.md:97]:
```suggestion
Just like you add livestreams to the Phoniebox, you can also add individual Podcasts or entire Podcast feeds to the Phoniebox.
```

File: documentation/builders/webapp/playlists-livestreams-podcasts.md [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+You are essentially assigning a folder (just like any other conventional folder) to your card representing the content of your playlist.
+
+## Livestreams
+
+In order to play radio livestreams on your Phoniebox, you use playlists to register your livestream and make it accessible.
+
+### Using livestream.txt playlist in Phoniebox
+
+1. [Follow the steps above](#using-m3u-playlist-in-phoniebox) to add a playlist to your Phoniebox (make sure you have created individual folders).
+2. When creating the playlist file, make sure it ends with `livestream.txt` instead of `*.m3u`.
```
Comment (s-martin) [inline documentation/builders/webapp/playlists-livestreams-podcasts.md:71]: We say here ending with, but in the examples that's the whole file name. If a file ending with livestream.txt suffix works, we should show that in the examples, too.

Same to podcasts.txt below.

File: documentation/builders/webapp/playlists-livestreams-podcasts.md [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+3. Download the MP3 and add them like normal songs to your Phoniebox. This makes them available offline.
+
+We will explain options 1 and 2 more closely.
+
+### Using podcast.txt playlist in Phoniebox
+
+1. [Follow the steps above](#using-m3u-playlist-in-phoniebox) to add a playlist to your Phoniebox (make sure you have created individual folders).
+2. When creating the playlist file, make sure it ends with `podcasts.txt` instead of `*.m3u`.
+3. Add links to your individual podcast episodes just like you would with songs in .m3u playlists
+4. As an alternative, you can provide a single RSS feed (XML). Phoniebox will expand the file and refer to all episodes listed within this file
```
Comment (s-martin) [inline documentation/builders/webapp/playlists-livestreams-podcasts.md:112]:
```suggestion
4. As an alternative, you can provide a single RSS feed (XML). Phoniebox will expand the file and refer to all episodes listed within this file.
```

File: documentation/builders/webapp/playlists-livestreams-podcasts.md [hunk 5]
```diff
@@ -0,0 +1,141 @@
+# Playlists, Livestreams and Podcasts
+
+By default, the Phoniebox represents music based on its metadata like album name, artist or song name. The hierarchy and order of songs is determined by their original definition. If you prefer a specific list of songs to be played, you can use playlists (files ending with `*.m3u`). Phoniebox also supports livestreams and podcasts (if connected to the internet) through playlists.
+
+## Playlists
+If you like the Phoniebox to play songs in the pre-defined order, you can use .m3u playlists.
+
+An .m3u playlist is a plain text file that contains a list of file paths or URLs to multimedia files. Each entry in the playlist represents a single song, and they are listed in the order in which they should be played.
```
Comment (AlvinSchiller) [inline documentation/builders/webapp/playlists-livestreams-podcasts.md:8]: Isn't it "a" playlist instead of "an" playlist? 
Occurs a few times throughout the text

File: documentation/builders/webapp/playlists-livestreams-podcasts.md [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+
+### Structure of an .m3u playlist
+An .m3u playlist is a simple text document with each song file listed on a separate line. Each entry is optionally preceded by a comment line that starts with a '#' symbol. The actual file paths or URLs of the media files come after the comment.
+
+### Creating an .m3u playlist
+
+1. You can create an .m3u playlist using a plain text editor.
+2. Open a new text file
```
Comment (AlvinSchiller) [inline documentation/builders/webapp/playlists-livestreams-podcasts.md:16]: You could just use `1.` for all entries, to make future changes easier.
Its the [markdown default](https://docs.github.com/de/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#lists) and gets replaces on rendering.

File: documentation/builders/webapp/playlists-livestreams-podcasts.md [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+3. [Optional] Start by adding a comment line to provide a description or notes about the playlist.
+4. On the following lines, list the file paths or URLs of the media files you want to include in the playlist, one per line. They must refer to true files paths on your Phoniebox. They can be relative or absolute paths.
+5. Save the file with the .m3u extensionm, e.g. `my_playlist.m3u`.
+
+```
+#EXTM3U
+Simone Sommerland/Die 30 besten Kindergartenlieder/08 - Pitsch, patsch, Pinguin.mp3
+Simone Sommerland/Die 30 besten Spiel- Und Bewegungslieder/12 - Das rote Pferd.mp3
+Bibi und Tina/bibi-tina-jetzt-in-echt-kinofilm-soundtrack/bibi-tina-jetzt-in-echt-kinofilm-soundtrack-7-ordinary-girl.mp3
```
Comment (AlvinSchiller) [inline documentation/builders/webapp/playlists-livestreams-podcasts.md:25]: Add an example for an absolute path?

File: documentation/builders/webapp/playlists-livestreams-podcasts.md (hunk 6 above)
Comment (AlvinSchiller) [inline documentation/builders/webapp/playlists-livestreams-podcasts.md:16]: Is it important to have the correct line endings for linux? Or does is also work if the file is created on Windows and then just copied?

File: documentation/builders/webapp/playlists-livestreams-podcasts.md [hunk 8]
```diff
@@ -0,0 +1,141 @@
...
+### Structure of an .m3u playlist
+An .m3u playlist is a simple text document with each song file listed on a separate line. Each entry is optionally preceded by a comment line that starts with a '#' symbol. The actual file paths or URLs of the media files come after the comment.
+
+### Creating an .m3u playlist
+
+1. You can create an .m3u playlist using a plain text editor.
+2. Open a new text file
+3. [Optional] Start by adding a comment line to provide a description or notes about the playlist.
+4. On the following lines, list the file paths or URLs of the media files you want to include in the playlist, one per line. They must refer to true files paths on your Phoniebox. They can be relative or absolute paths.
+5. Save the file with the .m3u extensionm, e.g. `my_playlist.m3u`.
```
Comment (AlvinSchiller) [inline documentation/builders/webapp/playlists-livestreams-podcasts.md:19]:
```suggestion
5. Save the file with the .m3u extension, e.g. `my_playlist.m3u`.
```

File: documentation/builders/webapp/playlists-livestreams-podcasts.md [hunk 9]
```diff
@@ -0,0 +1,141 @@
...
+
+```
+#EXTM3U
```
Comment (AlvinSchiller) [inline documentation/builders/webapp/playlists-livestreams-podcasts.md:22]: From the description its a simple m3u file, so better remove `#EXTM3U` to not confuse the user.
I think you would not want to create extended m3u data manually ^^

Do we read them anyway?

File: documentation/builders/webapp/playlists-livestreams-podcasts.md [hunk 10]
```diff
@@ -0,0 +1,141 @@
...
+Simone Sommerland/Die 30 besten Kindergartenlieder/08 - Pitsch, patsch, Pinguin.mp3
+Simone Sommerland/Die 30 besten Spiel- Und Bewegungslieder/12 - Das rote Pferd.mp3
+Bibi und Tina/bibi-tina-jetzt-in-echt-kinofilm-soundtrack/bibi-tina-jetzt-in-echt-kinofilm-soundtrack-7-ordinary-girl.mp3
+```
+
+### Using .m3u playlist in Phoniebox
```
Comment (AlvinSchiller) [inline documentation/builders/webapp/playlists-livestreams-podcasts.md:28]:
```suggestion
### Using .m3u playlists in Phoniebox
```

File: documentation/builders/webapp/playlists-livestreams-podcasts.md [hunk 11]
```diff
@@ -0,0 +1,141 @@
...
+
+The Phoniebox Web App handles the playlists in a way that it allows you to browse its content just like other songs. This means, you won't see the m3u playlist itself and instead, the individual items of the playlist. They also become actionable and you can select individual songs from it to play, or play the entire playlist.
+
+> [!NOTE]
+> Files ending with `.m3u` are treated as folder playlist. Regular folder processing is suspended and the playlist is build solely from the `.m3u` content. Only the alphabetically first `.m3u` is processed, others are ignored.
```
Comment (AlvinSchiller) [inline documentation/builders/webapp/playlists-livestreams-podcasts.md:33]:
```suggestion
> Files ending with `.m3u` are treated as folder playlist. Regular folder processing is suspended and the playlist is build solely from the `.m3u` content. Only the alphabetically first `.m3u` file is processed, others are ignored.
```

File: documentation/builders/webapp/playlists-livestreams-podcasts.md [hunk 12]
```diff
@@ -0,0 +1,141 @@
...
+
+Based on the note above, we suggest to use m3u playlists like this, especially if you like to manage multiple playlists.
+
+1. In the `audiofolders` directory (or any sub-directory), create a new folder.
+2. In this new folder, copy your .m3u playlist. Make sure the links to the respective songs are correct.
+3. Open the Web App. Under Library, select the Folder view and browse to the new folder you created.
```
Comment (AlvinSchiller) [inline documentation/builders/webapp/playlists-livestreams-podcasts.md:39]:
```suggestion
3. Open the Web App and go to `Library`. Select the `Folder` view and browse to the new folder you created.
```

File: documentation/builders/webapp/playlists-livestreams-podcasts.md [hunk 13]
```diff
@@ -0,0 +1,141 @@
...
+    │   └── playlist.m3u
+    └── lullabies-sleep-well
+        └── playlist.m3u
+```
+
+### Assigning a .m3u playlist to a card
+
+In the Phoniebox Web App, .m3u playlists do not show up as individual files. In order to assign a playlist to a card, do the following:
+
+1. [Follow the steps above](#using-m3u-playlist-in-phoniebox) to add a playlist to your Phoniebox (make sure you have created individual folders).
```
Comment (AlvinSchiller) [inline documentation/builders/webapp/playlists-livestreams-podcasts.md:56]: correlacting to the changes to the header above
```suggestion
1. [Follow the steps above](#using-m3u-playlists-in-phoniebox) to add a playlist to your Phoniebox (make sure you have created individual folders).
```

File: documentation/builders/webapp/playlists-livestreams-podcasts.md [hunk 14]
```diff
@@ -0,0 +1,141 @@
...
+2. In the Web App, open Cards tab and click the "+" button to add a new card.
```
Comment (AlvinSchiller) [inline documentation/builders/webapp/playlists-livestreams-podcasts.md:57]: Harmonized with the wording above.
```suggestion
2. Open the Web App and go to `Cards`. Click the "+" button to add a new card.
```

File: documentation/builders/webapp/playlists-livestreams-podcasts.md [hunk 15]
```diff
@@ -0,0 +1,141 @@
+# Playlists, Livestreams and Podcasts
+
+By default, the Phoniebox represents music based on its metadata like album name, artist or song name. The hierarchy and order of songs is determined by their original definition. If you prefer a specific list of songs to be played, you can use playlists (files ending with `*.m3u`). Phoniebox also supports livestreams and podcasts (if connected to the internet) through playlists.
```
Comment (AlvinSchiller) [inline documentation/builders/webapp/playlists-livestreams-podcasts.md:3]: Should we call it `Jukebox` or `Phoniebox` ? 
I think in most docs its `Jukebox` (legal actions for the name "Phoniebox" on mind?).

File: documentation/builders/webapp/playlists-livestreams-podcasts.md [hunk 16]
```diff
@@ -0,0 +1,141 @@
...
+```
+
+### Assigning a .m3u playlist to a card
+
+In the Phoniebox Web App, .m3u playlists do not show up as individual files. In order to assign a playlist to a card, do the following:
+
+1. [Follow the steps above](#using-m3u-playlist-in-phoniebox) to add a playlist to your Phoniebox (make sure you have created individual folders).
+2. In the Web App, open Cards tab and click the "+" button to add a new card.
+3. As a Jukebox action, select "Play music", then select "Select music".
+4. Next (in your your Library view), select the Folder view in the top right corner.
```
Comment (AlvinSchiller) [inline documentation/builders/webapp/playlists-livestreams-podcasts.md:59]:
```suggestion
4. Next (in your Library view), select the Folder view in the top right corner.
```

File: documentation/builders/webapp/playlists-livestreams-podcasts.md [hunk 17]
```diff
@@ -0,0 +1,141 @@
...
+
+You are essentially assigning a folder (just like any other conventional folder) to your card representing the content of your playlist.
+
+## Livestreams
+
+In order to play radio livestreams on your Phoniebox, you use playlists to register your livestream and make it accessible.
+
+### Using livestream.txt playlist in Phoniebox
+
+1. [Follow the steps above](#using-m3u-playlist-in-phoniebox) to add a playlist to your Phoniebox (make sure you have created individual folders).
```
Comment (AlvinSchiller) [inline documentation/builders/webapp/playlists-livestreams-podcasts.md:70]: correlacting to the changes to the header above
```suggestion
1. [Follow the steps above](#using-m3u-playlists-in-phoniebox) to add a playlist to your Phoniebox (make sure you have created individual folders).
```

File: documentation/builders/webapp/playlists-livestreams-podcasts.md [hunk 18]
```diff
@@ -0,0 +1,141 @@
...
+2. When creating the playlist file, make sure it call or at least ends with `livestream.txt` instead of `*.m3u` (Examples: `awesome-livestream.txt`, `livestream.txt`).
```
Comment (AlvinSchiller) [inline documentation/builders/webapp/playlists-livestreams-podcasts.md:71]:
```suggestion
2. When creating the playlist file, make sure it is called or at least ends with `livestream.txt` instead of `*.m3u` (Examples: `awesome-livestream.txt`, `livestream.txt`).
```

File: documentation/builders/webapp/playlists-livestreams-podcasts.md [hunk 19]
```diff
@@ -0,0 +1,141 @@
...
+3. Add links to your livestreams just like you would with songs in .m3u playlists
```
Comment (AlvinSchiller) [inline documentation/builders/webapp/playlists-livestreams-podcasts.md:72]:
```suggestion
3. Add links of your livestreams just like you would with songs in .m3u playlists
```

File: documentation/builders/webapp/playlists-livestreams-podcasts.md [hunk 20]
```diff
@@ -0,0 +1,141 @@
...
+http://channels.webradio.antenne.de/hits-fuer-kids
+```
+
+## Podcasts
+
+Just like you add livestreams to the Phoniebox, you can also add individual Podcasts or entire Podcast feeds to the Phoniebox.
+
+You have 3 options to play Podcasts
+
+1. Create a playlist and reference individual direct links to Podcast episodes (just like livestreams)
```
Comment (AlvinSchiller) [inline documentation/builders/webapp/playlists-livestreams-podcasts.md:101]:
```suggestion
1. Create a playlist and reference individual direct links to Podcast episodes (just like [livestreams](#livestreams))
```

File: documentation/builders/webapp/playlists-livestreams-podcasts.md [hunk 21]
```diff
@@ -0,0 +1,141 @@
...
+2. Provide a Podcast RSS feed
+3. Download the MP3 and add them like normal songs to your Phoniebox. This makes them available offline.
+
+We will explain options 1 and 2 more closely.
+
+### Using podcast.txt playlist in Phoniebox
+
+1. [Follow the steps above](#using-m3u-playlist-in-phoniebox) to add a playlist to your Phoniebox (make sure you have created individual folders).
```
Comment (AlvinSchiller) [inline documentation/builders/webapp/playlists-livestreams-podcasts.md:109]: correlacting to the changes to the header above
```suggestion
1. [Follow the steps above](#using-m3u-playlists-in-phoniebox) to add a playlist to your Phoniebox (make sure you have created individual folders).
```

File: documentation/builders/webapp/playlists-livestreams-podcasts.md [hunk 22]
```diff
@@ -0,0 +1,141 @@
...
+2. When creating the playlist file, make sure it ends with `podcasts.txt` instead of `*.m3u`. (Examples: `awesome-podcast.txt`, `podcast.txt`).
```
Comment (AlvinSchiller) [inline documentation/builders/webapp/playlists-livestreams-podcasts.md:110]: Harmonized with description above
```suggestion
2. When creating the playlist file, make sure it is called or at least ends with `podcasts.txt` instead of `*.m3u`. (Examples: `awesome-podcast.txt`, `podcast.txt`).
```

File: documentation/builders/webapp/playlists-livestreams-podcasts.md (hunk 8 above)
Comment (s-martin) [inline documentation/builders/webapp/playlists-livestreams-podcasts.md:19]:
```suggestion
5. Save the file with the .m3u extension, e.g. `my_playlist.m3u`.
```

File: documentation/builders/webapp/playlists-livestreams-podcasts.md (hunk 15 above)
Comment (Groovylein) [reply documentation/builders/webapp/playlists-livestreams-podcasts.md:3]: At the very beginning of the v2 README it is called `Phoniebox` ;)

Comment (AlvinSchiller) [reply documentation/builders/webapp/playlists-livestreams-podcasts.md:3]: True 😄 
I did now search in the v3 docs and both is often used, so i was wrong. 

Maybe we should define to only use one wording? `Phoniebox`?

Comment (pabera) [reply documentation/builders/webapp/playlists-livestreams-podcasts.md:3]: I am actually voting for `Jukebox`. Mainly for reasons @AlvinSchiller called out. We could then use `Phoniebox` only in discussions or in a few positions where it makes sense.

File: documentation/builders/webapp/playlists-livestreams-podcasts.md [hunk 23]
```diff
@@ -0,0 +1,142 @@
...
+1. You can create a .m3u playlist using a plain text editor.
+1. Open a new text file
+1. [Optional] Start by adding a comment line to provide a description or notes about the playlist.
+1. On the following lines, list the file paths or URLs of the media files you want to include in the playlist, one per line. They must refer to true files paths on your Jukebox. They can be relative or absolute paths.
+1. Save the file with the .m3u extension, e.g. `my_playlist.m3u`.
+
+```
+# Absolute
+/root/RPi-Jukebox-RFID/shared/audiofolders/Simone Sommerland/Die 30 besten Kindergartenlieder/08 - Pitsch, patsch, Pinguin.mp3
+/root/RPi-Jukebox-RFID/shared/audiofolders/Simone Sommerland/Die 30 besten Spiel- Und Bewegungslieder/12 - Das rote Pferd.mp3
```
Comment (AlvinSchiller) [inline documentation/builders/webapp/playlists-livestreams-podcasts.md:24]:
```suggestion
/home/<username>/RPi-Jukebox-RFID/shared/audiofolders/Simone Sommerland/Die 30 besten Kindergartenlieder/08 - Pitsch, patsch, Pinguin.mp3
/home/<username>/RPi-Jukebox-RFID/shared/audiofolders/Simone Sommerland/Die 30 besten Spiel- Und Bewegungslieder/12 - Das rote Pferd.mp3
```

File: documentation/builders/webapp/playlists-livestreams-podcasts.md [hunk 24]
```diff
@@ -0,0 +1,142 @@
...
+    │   └── playlist.m3u
+    └── lullabies-sleep-well
+        └── playlist.m3u
+```
+
+### Assigning a .m3u playlist to a card
+
+In the Jukebox Web App, .m3u playlists do not show up as individual files. In order to assign a playlist to a card, do the following:
+
+1. [Follow the steps above](#using-m3u-playlist-in-jukebox) to add a playlist to your Jukebox (make sure you have created individual folders).
```
Comment (AlvinSchiller) [inline documentation/builders/webapp/playlists-livestreams-podcasts.md:57]:
```suggestion
1. [Follow the steps above](#using-m3u-playlists-in-jukebox) to add a playlist to your Jukebox (make sure you have created individual folders).
```

File: documentation/builders/webapp/playlists-livestreams-podcasts.md [hunk 25]
```diff
@@ -0,0 +1,142 @@
...
+1. In the Web App, open `Cards` tab and click the `+` button to add a new card.
```
Comment (AlvinSchiller) [inline documentation/builders/webapp/playlists-livestreams-podcasts.md:58]: Harmonized with wording in line 40
```suggestion
1. Open the Web App. Under `Cards`, click the `+` button to add a new card.
```

File: documentation/builders/webapp/playlists-livestreams-podcasts.md [hunk 26]
```diff
@@ -0,0 +1,142 @@
...
+
+You are essentially assigning a folder (just like any other conventional folder) to your card representing the content of your playlist.
+
+## Livestreams
+
+In order to play radio livestreams on your Jukebox, you use playlists to register your livestream and make it accessible.
+
+### Using livestream.txt playlist in Jukebox
+
+1. [Follow the steps above](#using-m3u-playlist-in-jukebox) to add a playlist to your Jukebox (make sure you have created individual folders).
```
Comment (AlvinSchiller) [inline documentation/builders/webapp/playlists-livestreams-podcasts.md:71]:
```suggestion
1. [Follow the steps above](#using-m3u-playlists-in-jukebox) to add a playlist to your Jukebox (make sure you have created individual folders).
```

File: documentation/builders/webapp/playlists-livestreams-podcasts.md [hunk 27]
```diff
@@ -0,0 +1,142 @@
...
+1. When creating the playlist file, make sure it's called or at least ends with `livestream.txt` instead of `.m3u` (Examples: `awesome-livestream.txt`, `livestream.txt`).
+1. Add URLs of your livestreams just like you would add songs in `.m3u` playlists.
+
+You can now assign livestreams to cards [following the example](#assiging-a-m3u-playlist-to-a-card) of playlists.
```
Comment (AlvinSchiller) [inline documentation/builders/webapp/playlists-livestreams-podcasts.md:75]:
```suggestion
You can now assign livestreams to cards [following the example](#assigning-a-m3u-playlist-to-a-card) of playlists.
```

File: documentation/builders/webapp/playlists-livestreams-podcasts.md [hunk 28]
```diff
@@ -0,0 +1,142 @@
...
+```
+
+### Assigning a .m3u playlist to a card
+
+In the Jukebox Web App, .m3u playlists do not show up as individual files. In order to assign a playlist to a card, do the following:
+
+1. [Follow the steps above](#using-m3u-playlist-in-jukebox) to add a playlist to your Jukebox (make sure you have created individual folders).
+1. In the Web App, open `Cards` tab and click the `+` button to add a new card.
+1. As a Jukebox action, select "Play music", then select "Select music".
+1. Next (in your your `Library` view), select the `Folder` view in the top right corner.
```
Comment (AlvinSchiller) [inline documentation/builders/webapp/playlists-livestreams-podcasts.md:60]:
```suggestion
1. Next (in the `Library` view), select the `Folder` view in the top right corner.
```

File: documentation/builders/webapp/playlists-livestreams-podcasts.md [hunk 29]
```diff
@@ -0,0 +1,142 @@
...
+
+1. Create a playlist and reference individual direct URLs to Podcast episodes (just like [livestreams](#livestreams))
+1. Provide a Podcast RSS feed
+1. Download the MP3 and add them like normal songs to your Jukebox. This also makes them available offline.
+
+We will explain options 1 and 2 more closely.
+
+### Using podcast.txt playlist in Jukebox
+
+1. [Follow the steps above](#using-m3u-playlist-in-jukebox) to add a playlist to your Jukebox (make sure you have created individual folders).
```
Comment (AlvinSchiller) [inline documentation/builders/webapp/playlists-livestreams-podcasts.md:110]:
```suggestion
1. [Follow the steps above](#using-m3u-playlists-in-jukebox) to add a playlist to your Jukebox (make sure you have created individual folders).
```

**Review by s-martin — APPROVED** [review]
I could not try it yet, but it looks fine to me.

File: documentation/builders/webapp/playlists-livestreams-podcasts.md [hunk 30]
```diff
@@ -0,0 +1,142 @@
...
+The Jukebox Web App handles the playlists in a way that it allows you to browse its content just like other songs. This means, you won't see the m3u playlist itself and instead, the individual items of the playlist. They also become actionable and you can select individual songs from it to play, or play the entire playlist.
+
+> [!NOTE]
+> Files ending with `.m3u` are treated as folder playlist. Regular folder processing is suspended and the playlist is built solely from the `.m3u` content. Only the alphabetically first `.m3u` file is processed, others are ignored.
+
+Based on the note above, we suggest to use m3u playlists like this, especially if you like to manage multiple playlists.
+
+1. In the `audiofolders` directory (or any sub-directory), create a new folder.
+1. In this new folder, copy your .m3u playlist. Make sure the links to the respective songs are correct.
+1. Open the Web App. Under `Library`, select the `Folder` view and browse to the new folder you created.
```
Comment (AlvinSchiller) [inline documentation/builders/webapp/playlists-livestreams-podcasts.md:40]:
```suggestion
1. Open the `Library` tab in the Web App, select the `Folder` view and browse to the new folder you created.
```

**Review by AlvinSchiller — APPROVED** [review]

**Review by pabera — APPROVED** [review]

File: documentation/builders/webapp/playlists-livestreams-podcasts.md (hunk 30 above)
Comment (pabera) [reply documentation/builders/webapp/playlists-livestreams-podcasts.md:40]: I'd like to keep it that way. It's represents a step by step guide where `Open Web App` comes first.

Comment (AlvinSchiller) [reply documentation/builders/webapp/playlists-livestreams-podcasts.md:40]: ok, just though of harmonizing the wording with Line 58

**Comment by flipreverse** [comment]
I'm very sorry returning that late to this PR. Thanks for considering my feedback.

One thing I don't get: Why do I have to put every playlist in a separate folder? On my opinion, it is the default behavior of nearly every audio player out there. Simply select the playlist  as a file, and I'm done.
Moreover, moving my old CDs to the Phoniebox would have been easy as ripping the CD, and automatically creating a m3u playlist. This way I'll have to create a dedicated subfolder for **every** playlist created before moving the files to the Phoniebox.

**Comment by pabera** [comment]
That's a good point. It was a choice. This way, you can still choose an individual song of your given playlist.

How do you organize your ripped CD's? In my opinion, you'd still manage them in individual folders.

I agree, this could still be added as an option. I'll keep it in mind.

To avoid too much manual work with copying the playlists, you could use a bash script to do this for you. If you have all your playlists in a folder, this script should be sufficient

```
for file in *.m3u; do folder=$(basename "$file" .m3u); mkdir -p "$folder" && mv "$file" "$folder"; done
```
